semeio/jobs/ots/ots_vel_surface.py

```diff
@@ -0,0 +1,198 @@
+import numpy as np
+
+from scipy.interpolate import CloughTocher2DInterpolator
+
+import segyio
+from segyio import TraceField
+
+
+class OTSVelSurface(object):
+    def __init__(self, res_surface, vcube):
+        """
+        Create a surface where the timeshift can be calculated.
+
+        Read a reservoir
+        """
+
+        self._x = None
+        self._y = None
+        self._z = None
+
+        self._nx = None
+        self._ny = None
+
+        self._z3d = None
+        self._dt = None
+
+        self._mapReservoirSurfaceToVelocity(res_surface, vcube)
+
+    @property
+    def x(self):
+        return self._x
+
+    @property
+    def y(self):
+        return self._y
+
+    @property
+    def z(self):
+        return self._z
+
+    @property
+    def nx(self):
+        return self._nx
+
+    @property
+    def ny(self):
+        return self._ny
+
+    @property
+    def z3d(self):
+        return self._z3d
+
+    @property
+    def dt(self):
+        return self._dt
+
+    def __len__(self):
+        return len(self.x)
+
+    def __str__(self):
+
+        s = "x: " + str(self.x)
+        s += "\ny: " + str(self.y)
+        s += "\nz: " + str(self.z)
+        s += "\ndt=" + str(self.dt)
+        s += "\nz3d=" + str(self.z3d)
+        return s
+
+    def _read_velocity(self, vcube, cell_corners):
+        """
+        Read velocity from segy file. Upscale.
+        :param vcube:
+        :param cell_corners:
+        :return:
+        """
+
+        with segyio.open(vcube) as f:
+            x = np.empty(f.tracecount, dtype=np.float64)
+            y = np.empty(f.tracecount, dtype=np.float64)
+            dt = segyio.dt(f) / 1e6
+            nt = len(f.samples)
+
+            for i, h in enumerate(f.header):
+                scalar = h[TraceField.SourceGroupScalar]
+                if scalar < 0:
+                    scalar = -1.0 / scalar
+                x[i] = h[TraceField.CDP_X] * scalar
+                y[i] = h[TraceField.CDP_Y] * scalar
+
+            # TODO nx <-> ny??
```

Comment:
Is there an issue for this, or is it fixed?

Reply:
well, this comment comes from before and IMHO it works it as now.